src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -105,7 +106,9 @@ private void RenderBeginTagHelperScope(string tagName, bool selfClosing, IList<C
             // per call site, e.g. if the tag is on the view twice, there should be two IDs.
             _writer.WriteStringLiteral(tagName)
                    .WriteParameterSeparator()
-                   .WriteBooleanLiteral(selfClosing)
+                   .Write(nameof(TagMode))
```

Comment:
Does MVC generate a `using` for the `Microsoft.AspNet.Razor.Runtime.TagHelpers` namespace? If not use `typeof(TagMode).FullName`.